compiler-rt/lib/builtins/aarch64/sme-libc-mem-routines.S

```diff
@@ -238,7 +236,8 @@ END_COMPILERRT_OUTLINE_FUNCTION(__arm_sc_memcpy)
 
 DEFINE_COMPILERRT_FUNCTION_ALIAS(__arm_sc_memmove, __arm_sc_memcpy)
 
-
+// This version uses FP registers. Use this only on targets with them
+#if defined(__aarch64__) && __ARM_FP != 0
```

Comment:
```suggestion
#if __ARM_FP != 0
```

Reply:
I'm not sure; the previous version protected this with `#ifdef __aarch64__`, so I just kept that in place. Seemed odd to me as well.

Reply:
@keith-packard I noticed that your patch has already been approved. Could you please proceed with submitting it upstream at your earliest convenience?

Thank you for your contribution and for helping to improve the project.

Reply:
I've rebased today; what else would I need to do to "submit it upstream"?

Reply:
Nothing else needed - I will click the merge button once CI completes. This will squash the commits, but I think that should be fine since they are both quite small. Is that okay or would you prefer to have them separate in the commit history?

Reply:
squashed is just fine with me; I like to keep things split apart for review, but after that, I have no strong opinions.